packages/artifact/src/internal/upload-http-client.ts

```diff
@@ -219,16 +219,22 @@ export class UploadHttpClient {
     httpClientIndex: number,
     parameters: UploadFileParameters
   ): Promise<UploadFileResult> {
-    const totalFileSize: number = (await stat(parameters.file)).size
+    const fileStat: fs.Stats = await stat(parameters.file)
+    const totalFileSize = fileStat.size
+    // on Windows with mkfifo from MSYS2 stats.isFIFO returns false, so we check if running on Windows node and
+    // if the file has size of 0 to compensate
+    const isFIFO =
+      fileStat.isFIFO() || (process.platform === 'win32' && totalFileSize === 0)
```

Comment:
Changes look good to me! Would like to get one more pair of eyes on this before merging

@yacaovsnc could you quickly glance over this PR? The only slight concern I have is the `(process.platform === 'win32' && totalFileSize === 0)` block. I'm not sure if there are any other edge cases where on Windows the file size would be reported as 0  and we don't want to go with an in-memory buffer.

Reply:
I think this is 👍🏼 .  The logic here will make sure any file with size 0 to use the file on disk approach which should be safer compared to the in-memory buffer.